lib/src/realm_class.dart

```diff
@@ -164,22 +164,22 @@ class Realm {
     }
   }
 
-  bool get _isInTransaction => realmCore.getIsWritable(this);
+  /// Checks whether the `Realm` is in write transaction.
+  bool get isInTransaction => realmCore.getIsWritable(this);
 
   /// Synchronously calls the provided callback inside a write transaction.
   ///
   /// If no exception is thrown from within the callback, the transaction will be committed.
   /// It is more efficient to update several properties or even create multiple objects in a single write transaction.
   T write<T>(T Function() writeCallback) {
+    final transaction = Transaction._(this);
```

Comment:
I don't understand why we need the `Transaction` class. The change really seems only to move the beginWrite outside the try block. I would like to understand more so perhaps we could discuss. Added an agenda topic

Reply:
Eventually we may want an API where the transaction is handed to user code, like some of the other SDKs.

Reply:
Discussed offline, we're keeping the transaction approach - @blagoev do you have a preference - should I expose it publicly in this PR or should I open a separate ticket for it?

Reply:
I think lets do it in another pr. This one only fixes the nested writes.